src/test/feature-math.spec.ts

```diff
@@ -335,5 +338,21 @@ describe('feature-math', () => {
         expect(await contract.getCompare28(dictA, dictB)).toBe(true);
         expect(await contract.getCompare28(dictB, dictA)).toBe(true);
         expect(await contract.getCompare28(dictA, dictA)).toBe(false);
+
+        // Test advanced math operations
+        for (let num = 1n; num <= 100n; num++) {
+            expect(await contract.getLog2(num)).toBe(
+                BigInt(Math.floor(Math.log2(Number(num))))
+            );
+        }
+
+        for (let num = 1n; num <= 10n; num++) {
```

Comment:
Please add more tests for corner cases:
- `num` is 0
- `num` is near the maximum integer value, i.e. `[maxint - 99; maxint]`
- `num` is in the middle of the range of the positive integers, i.e. `[maxint/2 - 50, maxint/2 + 50]`